### DISCUSSION TITLE
tox code testing issues

### DISCUSSION THREAD

**Discussion by ollimeier** [fonttools/fontbakery#4221] [Q&A] [answered]
As written here:
https://font-bakery.readthedocs.io/en/stable/developer/source-contrib-guide.html#install-tox

I should be able to run tox. But when doing so, I get:
ERROR: tox config file (either pyproject.toml, tox.ini, setup.cfg) not found

The weird thing, the documentation says 'You can find the versions of the Python interpreter that are used for testing in the tox.ini file.', but there is no 'tox.ini' in the top folder.

Does anyone experiences the same? Any help is much appreciated.
Thanks a lot in advance.

**Comment by felipesanches** [comment, marked as answer]
We recently removed tox setup files, but forgot to update the documentation. I was not using it and apparently @miguelsousa also was not interested in using it. And I was not aware of other people relying on that. But yeah, I agree that it was a potentially useful feature for some users, so, since it seems that this is your case, I think we should bring it back.

**Reply by ollimeier** [reply]
@felipesanches You don't need to bring it back just for me. Just let me know, how you do the testing and I will follow your approach.

**Reply by ollimeier** [reply]
The question of me is (and the documentation): How do you run tests (if not via tox)?

**Reply by felipesanches** [reply]
I've been running them directly on the CI jobs. But I agree that it may be faster to run them locally and tox used to be one way of doing it. Now, without tox, one could invoke pytest manually.

**Reply by ollimeier** [reply]
When I run  `pytest` in the top-level folder via terminal I get the following error:
```
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: -n
  inifile: /Users/ollimeier/Documents/GitHub/olli/fontbakery/pyproject.toml
  rootdir: /Users/ollimeier/Documents/GitHub/olli/fontbakery
```

Is this related to:
https://github.com/googlefonts/fontbakery/blob/9045ee2254a09f0c8f7ec07a301ca58d3a4e5bd9/pyproject.toml#L18C2-L18C2

**Reply by felipesanches** [reply]
yeah... I'm not sure why it does not recognize the `-n` argument. I recall it used to work at some point. I've now searched for info on specific versions of pytest in which the flag was added (and if it was ever deprecated) but I could not yet find the answer.

For now, removing that "-n 8" makes is work. I'd like to be sure about what's going on before making any changes, though.
